roxie/ccd/ccdqueue.cpp

```diff
@@ -2434,6 +2428,51 @@ class RoxieSocketQueueManager : public RoxieReceiverBase
     }
 };
 
+class RoxieUdpSocketQueueManager : public RoxieSocketQueueManager
+{
+public:
+    RoxieUdpSocketQueueManager(unsigned snifferChannel, unsigned _numWorkers) : RoxieSocketQueueManager(_numWorkers)
+    {
+        int udpQueueSize = topology->getPropInt("@udpQueueSize", UDP_QUEUE_SIZE);
+        int udpSendQueueSize = topology->getPropInt("@udpSendQueueSize", UDP_SEND_QUEUE_SIZE);
+        int udpMaxSlotsPerClient = topology->getPropInt("@udpMaxSlotsPerClient", 0x7fffffff);
+        if (topology->getPropInt("@sendMaxRate", 0))
+        {
+            unsigned sendMaxRate = topology->getPropInt("@sendMaxRate");
+            unsigned sendMaxRatePeriod = topology->getPropInt("@sendMaxRatePeriod", 1);
+            bucket.setown(new TokenBucket(sendMaxRate, sendMaxRatePeriod, sendMaxRate));
+            throttledPacketSendManager.setown(new RoxieThrottledPacketSender(*bucket, maxPacketSize));
+        }
+
+        IpAddress snifferIp;
+        getChannelIp(snifferIp, snifferChannel);
+        if (udpMaxSlotsPerClient > udpQueueSize)
+            udpMaxSlotsPerClient = udpQueueSize;
+        unsigned serverFlowPort = topology->getPropInt("@serverFlowPort", CCD_SERVER_FLOW_PORT);
+        unsigned dataPort = topology->getPropInt("@dataPort", CCD_DATA_PORT);
+        unsigned clientFlowPort = topology->getPropInt("@clientFlowPort", CCD_CLIENT_FLOW_PORT);
+        unsigned snifferPort = topology->getPropInt("@snifferPort", CCD_SNIFFER_PORT);
+        receiveManager.setown(createReceiveManager(serverFlowPort, dataPort, clientFlowPort, snifferPort, snifferIp, udpQueueSize, udpMaxSlotsPerClient));
+        sendManager.setown(createSendManager(serverFlowPort, dataPort, clientFlowPort, snifferPort, snifferIp, udpSendQueueSize, fastLaneQueue ? 3 : 2, bucket));
+    }
+
+};
+
+class RoxieAeronSocketQueueManager : public RoxieSocketQueueManager
+{
+public:
+    RoxieAeronSocketQueueManager(unsigned _numWorkers) : RoxieSocketQueueManager(_numWorkers)
+    {
+        unsigned dataPort = topology->getPropInt("@dataPort", CCD_DATA_PORT);
```

Comment:
Roxie has never (yet) supported that. I do want to at least experiment with changing so that we do, passing endpoints rather than ips in all messages, though it's actually getting less significant as we move towards containers...

Reply:
I'll add a separate subtask for that work

Reply:
https://track.hpccsystems.com/browse/HPCC-22493